Silicon/NXP/iMXPlatformPkg/Drivers/GopNullDxe/GopNullDxe.c

```diff
@@ -0,0 +1,271 @@
+/** @file
+*
+*  Copyright (c) 2019 Microsoft Corporation. All rights reserved.
+*  Copyright 2018 NXP
+*
+*  This program and the accompanying materials
+*  are licensed and made available under the terms and conditions of the BSD License
+*  which accompanies this distribution.  The full text of the license may be found at
+*  http://opensource.org/licenses/bsd-license.php
+*
+*  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+*
+**/
+
+#include "GopNullDxe.h"
+
+CONST DISPLAY_TIMING mDisplayTiming = {
+  1920,  // Horizontal Resolution
+  1080,  // Vertical Resolution
+  32,    // Bits Per Pixel
+};
+
+STATIC VID_DEVICE_PATH mVidDevicePath = {
+  {
+    {
+      HARDWARE_DEVICE_PATH,
+      HW_VENDOR_DP,
+      {
+        (UINT8)sizeof (VENDOR_DEVICE_PATH),
+        (UINT8) ((sizeof (VENDOR_DEVICE_PATH)) >> 8),
+      }
+    },
+    {
+      0xa6b94ebe,
+      0x5ba3,
+      0x44b0,
+      { 0x95, 0x92, 0xdc, 0x04, 0x5e, 0xb8, 0xf8, 0x9e }
+    }
+  },
+  {
+    END_DEVICE_PATH_TYPE,
+    END_ENTIRE_DEVICE_PATH_SUBTYPE,
+    {
+      sizeof (EFI_DEVICE_PATH_PROTOCOL),
+      0
+    }
+  }
+};
+
+STATIC EFI_GRAPHICS_OUTPUT_MODE_INFORMATION VidGopModeInfo;
+STATIC EFI_GRAPHICS_OUTPUT_PROTOCOL_MODE VidGopMode;
+
+EFI_STATUS
+EFIAPI
+VidGopQueryMode (
+  IN  EFI_GRAPHICS_OUTPUT_PROTOCOL *This,
+  IN  UINT32 ModeNumber,
+  OUT UINTN *SizeOfInfo,
+  OUT EFI_GRAPHICS_OUTPUT_MODE_INFORMATION **Info
+  )
+{
+  EFI_GRAPHICS_OUTPUT_MODE_INFORMATION  *OutputMode;
+  EFI_STATUS                            Status;
+
+  if (ModeNumber != 0) {
+    DEBUG ((DEBUG_ERROR, "%a: Saw request to query mode %d\n",
+      __FUNCTION__, ModeNumber));
+    Status = EFI_INVALID_PARAMETER;
+    goto Exit;
+  }
+
+  OutputMode = (EFI_GRAPHICS_OUTPUT_MODE_INFORMATION *)
+    AllocatePool (sizeof (EFI_GRAPHICS_OUTPUT_MODE_INFORMATION));
+  if (OutputMode == NULL) {
+    Status = EFI_OUT_OF_RESOURCES;
+    goto Exit;
+  }
+
+  OutputMode->Version = 0;
+  OutputMode->HorizontalResolution = VidGopModeInfo.HorizontalResolution;
+  OutputMode->VerticalResolution = VidGopModeInfo.VerticalResolution;
+  OutputMode->PixelFormat = VidGopModeInfo.PixelFormat;
+  OutputMode->PixelsPerScanLine = VidGopModeInfo.HorizontalResolution;
+  *SizeOfInfo = sizeof (EFI_GRAPHICS_OUTPUT_MODE_INFORMATION);
+  *Info = OutputMode;
+
+  Status = EFI_SUCCESS;
+
+Exit:
+  return Status;
+}
+
+EFI_STATUS
+VidGopSetMode (
+  IN EFI_GRAPHICS_OUTPUT_PROTOCOL   *This,
+  IN UINT32                         ModeNumber
+  )
+{
+  EFI_STATUS  Status;
+
+  if (ModeNumber != 0) {
+    DEBUG ((DEBUG_ERROR, "%a: Saw request to set mode to %d\n",
+      __FUNCTION__, ModeNumber));
+    Status = EFI_UNSUPPORTED;
+    goto Exit;
+  }
+
+  Status = EFI_SUCCESS;
+
+Exit:
+  return Status;
+}
+
+EFI_STATUS
+VidGopBlt (

```

Comment:
I hope you're removing the other instances of this code and modifying them to use this driver?

Reply:
This driver is to help address https://github.com/ms-iot/imx-iotcore/issues/66
It is not meant to replace our current Gop drivers.